### PR TITLE
kube-up for GCE chooses master size based on number of nodes

### LINKED ISSUE #22239: kube-up.sh should use reasonable master size based on NUM_NODES
```
export NUM_NODES=1000
cluster/kube-up.sh
```
That results in an n1-standard-2 master, which is clearly undersized.  kube-up.sh should use a reasonable master instance type based on NUM_NODES.

We should also clarify https://github.com/kubernetes/kubernetes/blob/master/docs/admin/cluster-large.md to document reasonable master sizes for various numbers of nodes/cores.

### PR DESCRIPTION
**Pull request by gmarek**
Fixes #22239

cc @alex-mohr @zmerlynn

### CODE REVIEW

File: cluster/gce/config-common.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    suggested_master_size=8
+  fi
+  if [[ "${NUM_NODES}" -gt "250" ]]; then
+    suggested_master_size=16
+  fi
+  if [[ "${NUM_NODES}" -gt "500" ]]; then
+    suggested_master_size=32
+  fi
+  echo "${suggested_master_size}"
+}
```
Comment (wojtek-t) [inline cluster/gce/config-common.sh]: nit: can you remove this strange char at the end of line?

File: cluster/gce/config-default.sh [hunk 2]
```diff
@@ -16,11 +16,14 @@
...
 # gcloud multiplexing for shared GCE/GKE tests.
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..
+source "${KUBE_ROOT}/cluster/gce/config-common.sh"
+
 GCLOUD=gcloud
 ZONE=${KUBE_GCE_ZONE:-us-central1-b}
-MASTER_SIZE=${MASTER_SIZE:-n1-standard-2}
 NODE_SIZE=${NODE_SIZE:-n1-standard-2}
 NUM_NODES=${NUM_NODES:-3}
+MASTER_SIZE=${MASTER_SIZE:-n1-standard-$(get-master-size)}
```
Comment (wojtek-t) [inline cluster/gce/config-default.sh]: please move "MASTER"-related var before NODE-related (alphabetical order)

Comment (wojtek-t) [inline cluster/gce/config-default.sh]: never mind - discussed offline

**Comment by wojtek-t** [comment]
LGTM

File: cluster/gce/config-common.sh (hunk 1 above)
Comment (gmarek) [inline cluster/gce/config-common.sh]: Done.

**Comment by gmarek** [comment]
Simple rebase. Reapplying LGTM.

**Pull request merged**